src/AccountNumberParser.java

```diff
@@ -52,16 +53,127 @@
 
     private static Account createAccountFromEntry(String[] entry) {
         String numberString = "";
+        ArrayList<Integer> indexesOfInvalidCharacters = new ArrayList<>();
+        ArrayList<String> invalidNumberString = new ArrayList<>();
+        int index = 0;
 
         for (String number : entry) {
             if (numbersMap.containsKey(number)) {
                 numberString += numbersMap.get(number);
             } else {
+                indexesOfInvalidCharacters.add(index);
+                invalidNumberString.add(number);
                 numberString += Account.INVALID_CHARACTER_MARK;
             }
+
+            index++;
+        }
+
+        Account account = new Account(numberString);
+
+        ArrayList<String> allCorrectAccounts = findAllPossibleAccountNumbersForIllegalAccount(account, indexesOfInvalidCharacters, invalidNumberString);
+        account = validateAndSetStateOfAccount(account, allCorrectAccounts);
+
+        return account;
+    }
+
+    private static ArrayList<String> findAllPossibleAccountNumbersForIllegalAccount(
```

Comment:
1) jak parsujemy numer, to zgadywanie co może być pod  ? należy zrobić w osobnej klasie
2) informacja o indeksach zgadywanych numerów powinna znaleźć się w Account
3) szukanie poprawnego numeru powinno być w osobnej klasie - dzięki indeksom niepoprawnych znaków nie będziemy ich już podmieniać
4) Account powinno przechowywać wszystkie numery kont, które są poprawne
5) nie potrzebujemy state, według mnie, bo jesteśmy w stanie wyznaczy na podstawie ilości numerów i stanu walidacji